pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/spanconfigsqltranslatorccl_test: TestDataDriven failed

pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/spanconfigsqltranslatorccl_test.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15515793?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15515793?buildTab=artifacts#/) on master @ [6d96126cf0769565e4756165888086a43db3e7ee](https://github.com/cockroachdb/cockroach/commits/6d96126cf0769565e4756165888086a43db3e7ee):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestDataDriven (4m57s)
	TestDataDriven/3node (4m57s)
	TestDataDriven/3node/partition/multi_col_list_partitioning_default_default_subpartitioned (4m57s)

```
Stack: 

```
goroutine 525297 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0x413f468000, {0x5500ca0?, 0xb72dfc0?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0x628
* runtime/pprof.Do({0x77c3de8?, 0xb72dfc0?}, {{0x40009cfc00?, 0x4136afcd50?, 0x4031e0bf68?}}, 0x4031e0bfb8)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x60
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 18230
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x44c
* 
* goroutine 474520 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0x40075af5f0, {0x77c3f38, 0x4006f15680})
* 	GOROOT/src/database/sql/sql.go:1246 +0x80
* created by database/sql.OpenDB in goroutine 81
* 	GOROOT/src/database/sql/sql.go:824 +0x140
* 
* goroutine 496538 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:253 +0x9c
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 63848
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0x64
* 
* goroutine 474527 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0x4034344000, {0x77c3f38, 0x405688c230})
* 	GOROOT/src/database/sql/sql.go:1246 +0x80
* created by database/sql.OpenDB in goroutine 81
* 	GOROOT/src/database/sql/sql.go:824 +0x140
* 
* goroutine 456332 [sync.Cond.Wait, 1 minutes]:
* sync.runtime_notifyListWait(0x41a0ab40a8, 0x12a)
* 	GOROOT/src/runtime/sema.go:569 +0x154
* sync.(*Cond).Wait(0x41a0ab4098)
* 	GOROOT/src/sync/cond.go:70 +0xcc
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:369
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0x41a0ab4000, {0x5500ca0?, 0xb72dfc0?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0x628
* runtime/pprof.Do({0x77c3de8?, 0xb72dfc0?}, {{0x40009cfc00?, 0x4195210570?, 0x40329fc768?}}, 0x40329fc7b8)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x60
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 455597
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x44c
* 
* goroutine 474399 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0x414063dad0, {0x77c3f38, 0x41451c4230})
* 	GOROOT/src/database/sql/sql.go:1246 +0x80
* created by database/sql.OpenDB in goroutine 81
* 	GOROOT/src/database/sql/sql.go:824 +0x140
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39228

pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/spanconfigsqltranslatorccl_test.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15518475?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15518475?buildTab=artifacts#/) on master @ [d2a9f45f1c9c8201bd1c32179e01eb75253ffb22](https://github.com/cockroachdb/cockroach/commits/d2a9f45f1c9c8201bd1c32179e01eb75253ffb22):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestDataDriven (4m57s)
	TestDataDriven/3node (4m57s)
	TestDataDriven/3node/repartition/multi_col_range_partitioning_maxvalue_to_multi_col_range_partitioning (4m31s)

```
Stack: 

```
goroutine 700314 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:253 +0x9c
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 343224
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:248 +0x64
* 
* goroutine 659475 [sync.Cond.Wait, 1 minutes]:
* sync.runtime_notifyListWait(0x40a85f80a8, 0x98)
* 	GOROOT/src/runtime/sema.go:569 +0x154
* sync.(*Cond).Wait(0x40a85f8098)
* 	GOROOT/src/sync/cond.go:70 +0xcc
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:369
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0x40a85f8000, {0x5500ca0?, 0xb72dfc0?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0x628
* runtime/pprof.Do({0x77c3de8?, 0xb72dfc0?}, {{0x4000154820?, 0x40a74bc450?, 0x4027aa9768?}}, 0x4027aa97b8)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x60
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 659362
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x44c
* 
* goroutine 662188 [sync.Cond.Wait, 1 minutes]:
* sync.runtime_notifyListWait(0x40e5f5e0a8, 0x8e)
* 	GOROOT/src/runtime/sema.go:569 +0x154
* sync.(*Cond).Wait(0x40e5f5e098)
* 	GOROOT/src/sync/cond.go:70 +0xcc
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:369
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0x40e5f5e000, {0x5500ca0?, 0xb72dfc0?})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:643 +0x628
* runtime/pprof.Do({0x77c3de8?, 0xb72dfc0?}, {{0x4000154820?, 0x48dd28?, 0x0?}}, 0x40b71a6fb8)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:562 +0x60
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 660289
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:561 +0x44c
* 
* goroutine 667041 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0x407fa161a0, {0x77c3f38, 0x40dd377a90})
* 	GOROOT/src/database/sql/sql.go:1246 +0x80
* created by database/sql.OpenDB in goroutine 317178
* 	GOROOT/src/database/sql/sql.go:824 +0x140
* 
* goroutine 667027 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0x408e5dbc70, {0x77c3f38, 0x40dd3778b0})
* 	GOROOT/src/database/sql/sql.go:1246 +0x80
* created by database/sql.OpenDB in goroutine 317178
* 	GOROOT/src/database/sql/sql.go:824 +0x140
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/BUILD.bazel
+++ b/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/BUILD.bazel
@@ -2,6 +2,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_test")
 
 go_test(
     name = "spanconfigsqltranslatorccl_test",
+    size = "large",
     srcs = [
         "datadriven_test.go",
         "main_test.go",

